docker/mongodb-kubernetes-tests/tests/search/search_community_tls.py

```diff
@@ -0,0 +1,324 @@
+        yaml_fixture("search-minimal.yaml"),
+        namespace=namespace,
+    )
+
+    # Add TLS configuration to MongoDBSearch
+    if "spec" not in resource:
+        resource["spec"] = {}
+
+    resource["spec"]["security"] = {"tls": {"enabled": True, "certificateKeySecretRef": {"name": MDBS_TLS_SECRET_NAME}}}
+
+    return resource
+
+
+@mark.e2e_search_community_tls
+def test_install_operator(namespace: str, operator_installation_config: dict[str, str]):
+    operator = get_default_operator(namespace, operator_installation_config=operator_installation_config)
+    operator.assert_is_running()
+
+
+@mark.e2e_search_community_tls
+def test_install_secrets(namespace: str, mdbs: MongoDBSearch):
+    # Create user password secrets
+    create_or_update_secret(namespace=namespace, name=f"{USER_NAME}-password", data={"password": USER_PASSWORD})
+    create_or_update_secret(
+        namespace=namespace, name=f"{ADMIN_USER_NAME}-password", data={"password": ADMIN_USER_PASSWORD}
+    )
+    create_or_update_secret(
+        namespace=namespace, name=f"{mdbs.name}-{MONGOT_USER_NAME}-password", data={"password": MONGOT_USER_PASSWORD}
+    )
+
+
+@mark.e2e_search_community_tls
+def test_install_tls_secrets_and_configmaps(
+    namespace: str, mdbc: MongoDBCommunity, mdbs: MongoDBSearch, ca_certificate
+):
+    mongodb_certs = generate_server_certificate(ca_certificate[0], ca_certificate[1], namespace, mdbc.name, "mdbc")
+    mongodb_tls_cert_data = {"tls.crt": mongodb_certs["cert"], "tls.key": mongodb_certs["key"]}
+    create_or_update_secret(
+        namespace=namespace, name=TLS_SECRET_NAME, data=mongodb_tls_cert_data, type="kubernetes.io/tls"
+    )
+
+    search_certs = generate_server_certificate(ca_certificate[0], ca_certificate[1], namespace, mdbs.name, "mdbs")
+    search_tls_cert_data = {"tls.crt": search_certs["cert"], "tls.key": search_certs["key"]}
+    create_or_update_secret(
+        namespace=namespace, name=MDBS_TLS_SECRET_NAME, data=search_tls_cert_data, type="kubernetes.io/tls"
+    )
+
+    ca_data = {"ca.crt": ca_certificate[2].read_text(encoding="utf-8")}  # Read CA cert from file
+    create_or_update_configmap(namespace=namespace, name=TLS_CA_CONFIGMAP_NAME, data=ca_data)
+
+
+@mark.e2e_search_community_tls
+def test_create_database_resource(mdbc: MongoDBCommunity):
+    mdbc.update()
+    mdbc.assert_reaches_phase(Phase.Running, timeout=1000)
+
+
+@mark.e2e_search_community_tls
+def test_create_search_resource(mdbs: MongoDBSearch):
+    mdbs.update()
+    mdbs.assert_reaches_phase(Phase.Running, timeout=300)
+
+
+@mark.e2e_search_community_tls
+def test_wait_for_community_resource_ready(mdbc: MongoDBCommunity):
+    mdbc.assert_reaches_phase(Phase.Running, timeout=1800)
+
+
+@mark.e2e_search_community_tls
+def test_validate_tls_connections(mdbc: MongoDBCommunity, mdbs: MongoDBSearch, namespace: str, ca_certificate):
+    ca_file_path = str(ca_certificate[2])
+
+    with pymongo.MongoClient(
```

Comment:
can we extend SearchTester to handle TLS? it's already using MongoTester which does those operations already. SearchTester inherits from MongoTester which has `assert_connectivity` method and I believe MongoTester handles TLS already

Reply:
movies_search_helper.assert_connectivity will be available just like that

Reply:
I agree with what you're saying, but I thought using SearchTester for that might be confusing - we create it with the mongod connection string so we can use its assert_connectivity method here as it's already supporting tls, but we need the additional connection string for testing mongot connectivity. Could you elaborate a bit on the changes you envision for SearchTester?